sdk/go/pulumi/types.go

```diff
@@ -551,6 +551,29 @@ func UnsafeUnknownOutput(deps []Resource) Output {
 	return output
 }
 
+// UnsafeAwaitedOutput is an output from a Pulumi function or resource that has been resolved.
+//
+// This is a low level API and should be used with care.
+type UnsafeAwaitedOutput struct {
+	Value        interface{} // The value of the output. If unknown (in a dry-run), the value will be nil.
+	Known        bool        // True if the value is known.
+	Secret       bool        // True if the value is a secret.
+	Dependencies []Resource  // The resources that this output depends on.
+}
+
+// UnsafeAwaitOutput blocks until the output is resolved and returns the resolved value and
+// metadata.
+func UnsafeAwaitOutput(o Output) (UnsafeAwaitedOutput, error) {
+	value, known, secret, deps, err := o.getState().await(context.Background())
```

Comment:
Are there any cases where it'd be useful to accept a `context.Context`?

Since this is an advanced API, I wonder if we should go ahead and accept that with this function, so we don't have to add a `UnsafeAwaitOutputWithContext` variant in the future.

```suggestion
func UnsafeAwaitOutput(ctx context.Context, o Output) (UnsafeAwaitedOutput, error) {
	value, known, secret, deps, err := o.getState().await(ctx)
```